numpy/polynomial/chebyshev.py

```diff
@@ -826,6 +826,9 @@ def chebpow(c, pow, maxpower=16):
     array([15.5, 22. , 16. , ..., 12.5, 12. ,  8. ])
 
     """
+    # note: this is more efficient than `pu._pow(chebmul, c1, c2)`, as it
+    # avoids converting between z and c series repeatedly
+
```

Comment:
The `_zseries_mul` is just convolution.

Reply:
Right, but calling `chebmul` in power results in:
```
c = c_in
z = ctoz(c)
z = zmul(z, c_in)
c = ztoc(z)
z = ctoz(c)
z = zmul(z, c_in)
c = ztoc(z)
...
instead of
```
c = c_in
z = ctoz(c)
z = zmul(z, c_in)
z = zmul(z, c_in)
c = ztoc(z)
...

Reply:
Don't call `chebmul` :) Convert to zseries and use `convolve`.

Reply:
If I do that, `as_series` gets called twice, which is something I was trying to avoid

Reply:
```
        zs = _cseries_to_zseries(c)
        prd = pu._pow(np.convolve, zs, pow, maxpow)
        return _zseries_to_cseries(prd)
```

Reply:
`_cseries_to_zseries(c)` will fail if `c` is not already an array. You need:
```
        [c] = pu.as_series([c])
        zs = _cseries_to_zseries(c)
        prd = pu._pow(np.convolve, zs, pow, maxpow)   # calls as_series a second time
        return _zseries_to_cseries(prd)
```
In the long run, the fix is to only call `as_series` in the convenience functions, but that sounds like work for a later CR.

Reply:
Hmm, `as_series` should not be too slow for a single array of valid type.  That function could maybe use a look to see if it could be sped up, the single argument version could follow a fast path. Be interesting to see what timing shows, but speed was not big consideration for the polynomial package, especially as it was designed so it would work with object coefficients (Decimal, Rational, etc.).

Reply:
`as_series` still makes a pointless copy. If efficiency isn't a concern, then maybe we should just merge #11816

Reply:
I don't feel it matters much one way or the other, just arguing for the extra simplification. Probably the biggest speedup would be to improve the algorithm with divide and conquer, at least for the bigger powers. The whole `as_series` thing could probably use a rethink as it gets called in all the functions.